libs/api_common/windows_platform_common.cpp

```diff
@@ -6,60 +6,351 @@
 #include "api_internal.h"
 #include "api_common.hpp"
 #include "crab_verifier_wrapper.hpp"
+#include "device_helper.hpp"
 #include "ebpf_api.h"
 #include "ebpf_nethooks.h"
+#include "ebpf_protocol.h"
+#include "ebpf_serialize.h"
 #include "helpers.hpp"
 #include "map_descriptors.hpp"
 #include "platform.hpp"
 #include "spec_type_descriptors.hpp"
 #include "utilities.hpp"
 #include "windows_program_type.h"
 #include "windows_platform.hpp"
+#include "um_registry_helper.h"
+
+#define GET_PROGRAM_INFO_REPLY_BUFFER_SIZE 2048
+
+static thread_local ebpf_handle_t _program_under_verification = ebpf_handle_invalid;
+
+// TODO: Issue #1231 Change to using HKEY_LOCAL_MACHINE
+static HKEY _root_registry_key = HKEY_CURRENT_USER;
+
+extern bool use_ebpf_store;
+
+struct guid_compare
+{
+    bool
+    operator()(const GUID& a, const GUID& b) const
+    {
+        return (memcmp(&a, &b, sizeof(GUID)) < 0);
+    }
+};
+
+struct _ebpf_program_info_deleter
+{
+    void
+    operator()(_In_ _Post_invalid_ ebpf_program_info_t* program_info)
+    {
+        ebpf_program_info_free(program_info);
+    }
+};
+
+static void
+_ebpf_program_descriptor_free(_In_opt_ EbpfProgramType* descriptor)
+{
+    EBPF_LOG_ENTRY();
+    if (descriptor == nullptr) {
+        EBPF_RETURN_VOID();
+    }
+
+    ebpf_free((void*)descriptor->context_descriptor);
+    ebpf_free((void*)descriptor->platform_specific_data);
+
+    delete descriptor;
+
+    EBPF_RETURN_VOID();
+}
+
+struct EbpfProgramType_deleter
+{
+    void
+    operator()(_In_ _Post_invalid_ EbpfProgramType* descriptor)
+    {
+        _ebpf_program_descriptor_free(descriptor);
+    }
+};
+
+// Thread local cache for program information queried from execution context.
+typedef std::unique_ptr<ebpf_program_info_t, _ebpf_program_info_deleter> ebpf_program_info_ptr_t;
+static thread_local std::map<ebpf_program_type_t, ebpf_program_info_ptr_t, guid_compare> _program_info_cache;
+
+// Thread local cache for program descriptor queried from execution context.
+typedef std::unique_ptr<EbpfProgramType, EbpfProgramType_deleter> ebpf_program_descriptor_ptr_t;
+static thread_local std::map<ebpf_program_type_t, ebpf_program_descriptor_ptr_t, guid_compare>
+    _program_descriptor_cache;
+
+// Global cache for the program and section information queried from eBPF store.
+static std::map<ebpf_program_type_t, EbpfProgramType, guid_compare> _windows_program_types;
+static std::vector<ebpf_section_definition_t> _windows_section_definitions;
+static std::map<ebpf_program_type_t, ebpf_program_info_t, guid_compare> _windows_program_information;
+
+void
+set_program_under_verification(ebpf_handle_t program)
+{
+    _program_under_verification = program;
+}
+
```

Comment:
[Minor] What is the reason for the refactoring of simply moving function between cpp files?

Reply:
I moved some other logic also (the functions which use `_program_under_verification` variable) to this file. Hence used these functions also to this file (as it is no longer needed in the other file)